### DISCUSSION TITLE
PMDG CDU Clr Button strange behaviour

### DISCUSSION THREAD

**Discussion by ksi9302** [EvenAR/node-simconnect#105] [Q&A] [answered]
Hi everyone,

Firstly thanks a lot for the great library.
Thanks to the library, I've made a full working CDU app for the recently released PMDG 77W for MSFS.
However, It's all working as intended except for the Clr button..

Basically, all other buttons are pressed for a short period, however clr button always gets pressed for long period resulting in clearing out all messages..

```
handle.transmitClientEvent(
        SimConnectConstants.OBJECT_ID_USER,
        MyEventID.CDU_CLR,
        1,
        0,
        EventFlag.EVENT_FLAG_GROUPID_IS_PRIORITY
    );
```

I'm basically using the same structure from [this](https://github.com/EvenAR/node-simconnect/blob/master/samples/typescript/pmdgCduScreen.ts) example.

Would be great if anyone has a clue on what's happening..

-- Edit --

I've tried using setClientData instead of transmitClientEvent, result is the same..

**Comment by ksi9302** [comment, marked as answer]
Okay, answering my own question.

TLDR;

Send mouse down action then mouse release action.
```
    handle.transmitClientEvent(
      SimConnectConstants.OBJECT_ID_USER,
      MyEventID.CDU_CLR,
      0x20000000,
      0,
      EventFlag.EVENT_FLAG_GROUPID_IS_PRIORITY
    );

    handle.transmitClientEvent(
      SimConnectConstants.OBJECT_ID_USER,
      MyEventID.CDU_CLR,
      0x00020000,
      0,
      EventFlag.EVENT_FLAG_GROUPID_IS_PRIORITY
    );
```

While desperately trying to find a clue, I've noticed that a parameter can be one of the MOUSE_FLAG which simulates mouse actions. // see SDK.h

Since buttons like Clr, and Menu has special function when long pressed, the default parameter 1 will act as a long press action for them.
In order to make it a single action, you should send a mouse release action followingly.

**Reply by EvenAR** [reply]
Thanks for sharing the solution 🙌 Out of curiosity, what kind of application are you making?

**Reply by ksi9302** [reply]
A CDU app for MSFS PMDG 77W with Electron serving both local server and web connected with socket! :) Should be available for free on Flightsim.to shortly!
